internal/decoder/decoder.go

```diff
@@ -66,7 +66,7 @@ func varsPathContext(mod *state.Module) (*decoder.PathContext, error) {
 		Files:            make(map[string]*hcl.File, 0),
 	}
 
-	for _, origin := range mod.RefOrigins {
+	for _, origin := range mod.VarsRefOrigins {
```

Comment:
Oops, I didn't realize we were already using the _wrong_ references here. 🙈  I think that LOC `74-78` can be deleted in that case.

Reply:
That's why I thought I had to merge them, but it's much cleaner with new fields on `state.Module`.